src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/RecommenderPackageFeed.cs

```diff
@@ -193,8 +210,20 @@ public async Task<IPackageSearchMetadata> GetPackageMetadataAsync(PackageIdentit
                 // and failing that we go to the network
                 packageMetadata = await _metadataProvider.GetPackageMetadataAsync(identity, includePrerelease, cancellationToken);
             }
-            return packageMetadata;
+            return new RecommendedPackageSearchMetadata(packageMetadata, VersionInfo);
         }
 
+        private class IdentityIdEqualityComparer : IEqualityComparer<IPackageSearchMetadata>
+        {
```

Comment:
A pattern I prefer is to make a private constructor with a static instance to ensure that no one makes extra instances of this

```suggestion
        {
            public static IdentityIdEqualityComparer Instance = new IdentityIdEqualityComparer();
            
            private IdentityIdEqualityComparer()
            {
            }
```

Then line 26 can go away and line 107 can use the static singleton

Reply:
I like this a lot, thanks!